lib/http/2/compressor.rb

```diff
@@ -513,10 +503,9 @@ def string(buf)
         huffman = (buf.readbyte(0) & 0x80) == 0x80
         len = integer(buf, 7)
         str = buf.read(len)
-        str.bytesize == len or raise CompressionError.new("string too short")
-        huffman and str = Huffman.new.decode(Buffer.new(str))
-        str = str.force_encoding('utf-8')
-        str
+        str.bytesize == len || fail(CompressionError, 'string too short')
+        huffman && str = Huffman.new.decode(Buffer.new(str))
```

Comment:
This is a good point - I've converted all the `condition && do_something` cases to `do_something if condition`